src/fidesops/api/v1/endpoints/masking_endpoints.py

```diff
@@ -30,7 +30,8 @@ def mask_value(masking_strategy: PolicyMaskingSpec, value: str) -> MaskingAPIRes
             masking_strategy.strategy, masking_strategy.configuration
         )
         logger.info(f"Starting masking with strategy {masking_strategy.strategy}")
-        masked_value = strategy.mask(value)
+        # fixme: generate secrets, but what privacy request id to use?
```

Comment:
Since we call `mask()` outside the context of a privacy request, how should we handle scoping the secrets? They are typically scoped by privacy request id. This is also complicated by the fact that currently, status quo is calling cache methods directly on the `PrivacyRequest` model.

Reply:
This decision will help determine:
1. If we should require a `privacy_request_id` in `mask()`. 
2. If we should require the caller of `mask()` to generate/cache needed secrets before calling `mask()`
3. If we should have the masking strategy generate/cache secrets if privacy_request_id is None

Reply:
This API is meant to act as a standalone masking API, so it shouldn't rely on a privacy request. I think we have two options here:
1. Require the caller supply all secrets for the masking (e.g. keys, peppers, nonces), which would allow deterministic masking
2. Generate all secrets randomly, which would make it non-deterministic

Since we don't have a very specific use case for this API yet (I imagine it being used to enable masking-as-a-service use cases in the future) I'd vote for option #2 as it is simple and easy to document and understand. It would therefore be a way to "test out" the various strategies before enabling them in an erasure policy.

Reply:
Regarding option 2, does this mean random within a given privacy request, i.e. the `key` for `HMAC` will be different for every privacy request?

Reply:
answer from Neville- yes, the keys should be generated randomly on each invocation of `mask()`, so we won't need to cache anything for now.